src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeter.kt

```diff
@@ -76,50 +109,83 @@ class ActionMeter(
         observation: (T) -> JsonObject?
     ): T {
         val start = clock.instant()
+        val actionMetricBuilder = ActionMetric.Builder(
```

Comment:
This uses dangerous defaults like `result: OK` and `duration: ZERO`. Perhaps we could avoid fake values only to hope they'll be mutated later?
The hardship comes from the fact that we're building an `ActionMetric` for both the `try` block and `catch` block at the same time. It might be worth sacrificing this forced reuse to gain safety.

So we gotta do one of the following:
* avoid dangerous defaults in the builder, stop reusing the builder for both happy and unhappy paths
* accept dangerous defaults and make them actual builder defaults instead of ad hoc secondary builder constructor